src/Security/Authentication/Certificate/src/CertificateValidationCache.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Microsoft.AspNetCore.Authentication.Certificate
+{
+    /// <summary>
+    /// MemoryCache based implementation used to store <see cref="AuthenticateResult"/> results after the certificate has been validated
+    /// </summary>
+    public class CertificateValidationCache : ICertificateValidationCache
+    {
+        private MemoryCache _cache;
+
+        public CertificateValidationCache()
+        {
+             _cache = new MemoryCache(new MemoryCacheOptions { SizeLimit = CacheSize });
```

Comment:
Aren't you reading CacheSize before anyone has a chance to change it? Should this be using `IOptions<CertificateValidationCache>`?

Reply:
Well I expected it to be more configured by derivation, MyCache : CertificateValidationCache and they change the properties

Reply:
Do we really need a top level options for this?

Reply:
> Well I expected it to be more configured by derivation, MyCache : CertificateValidationCache and they change the properties

That's not a pattern we use anywhere else is it?

> Do we really need a top level options for this?

That's the standard pattern.

Reply:
Well I mostly meant that I didn't think tweaking the cache slightly was going to be the main idea, I thought perhaps mistakenly that we would expect people to just implement their own cache and drop it in mostly.  But if we are going to add first class sugar, it makes sense to expose options